homeassistant/helpers/template.py

```diff
@@ -1139,8 +1153,12 @@ def format_state(self, rounded: bool, with_unit: bool) -> str:
             state = async_rounded_state(self._hass, self._entity_id, self._state)
         else:
             state = self._state.state
-        if with_unit and (unit := self._state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)):
+        if with_unit and (
+            (unit := self._state.attributes.get(ATTR_UNIT_OF_MEASUREMENT))
+            or (unit := self._get_translated_unit_of_measurement("en"))
```

Comment:
Why is Snglish hardcoded, shouldn't we use `hass.config.language` ? Also, how does the logic work here, is the unit of measurement falsy for translated units?